independent-projects/arc/processor/src/main/java/io/quarkus/arc/processor/Methods.java

```diff
@@ -505,6 +509,20 @@ public boolean test(MethodInfo method) {
                 // Do not skip default methods - public non-abstract instance methods declared in an interface
                 return false;
             }
+
+            List<Type> parameters = method.parameters();
+            if (!parameters.isEmpty() && (beanArchiveIndex != null)) {
+                for (Type type : parameters) {
+                    ClassInfo parameterClassInfo = beanArchiveIndex.getClassByName(type.name());
+                    if (parameterClassInfo == null) {
+                        continue; // hope for the best
+                    }
+                    if (!Modifier.isPublic(parameterClassInfo.flags())) {
```

Comment:
Hm, I'm not really sure this is the correct fix. If we go this way then we should probably at least log a warning. But I'm not quite sure that only public param classes are acceptable. What about package-private param classes? These should be probably OK because the generated subclass has the same package. 

@manovotn Do you remember what Weld does in similar cases?

Reply:
I thought about this as well. We could I guess check if the package of the class to which the method belongs is the same as the package of the parameter's class if that class is package-private.

But I'll hold off until we know what Weld does

Reply:
@mkouba @geoand sorry for the delay, below are my thoughts on this.

> These should be probably OK because the generated subclass has the same package.

Yea, this should be fine unless we have some edge cases where we have to create subclasses into dedicated (different) packages.

> Do you remember what Weld does in similar cases?

I don't remember so I wrote a similar test to what Georgios has here and Weld executes that with interception just fine.

Reply:
Thanks for the info @manovotn

OK, so basically Weld allows parameters whose class is package private parameters and use the same package, correct?

I'll update the PR accordingly

Reply:
> I don't remember so I wrote a similar test to what Georgios has here and Weld executes that with interception just fine.

And for a private parameter type?

Ok, so I think that we should ignore and log a warning only for a `private` parameter type. WDYT?

Reply:
> And for a private parameter type?

That requires a somewhat awkward setup to be able to reference that type and pass it as an arg. into the intercepted method. But yes, Weld will allow you to do that as well and interception will work.

Reply:
Hmm, actually we could try to fallback to reflection as well... But let's create a separate issue for that.

Reply:
I think what I pushed now covers the common cases - it certainly handles the original issue :)